runimagej.py

```diff
@@ -0,0 +1,359 @@
+]
+
+IMAGE_TYPES = [
+    "net.imagej.Dataset",
+    "net.imagej.display.DataView",
+    "net.imagej.display.DatasetView",
+    "net.imagej.display.ImageDisplay",
+    "net.imglib2.IterableInterval",
+    "net.imglib2.RandomAccessibleInterval",
+    "net.imglib2.img.*Img"
+]
+
+INTEGER_TYPES = [
+    "byte",
+    "int",
+    "long",
+    "short",
+    "java.lang.Byte",
+    "java.lang.Integer",
+    "java.lang.Long",
+    "java.lang.Short",
+    "java.math.BigInteger"
+]
+
+TEXT_TYPES = [
+    "char",
+    "java.lang.Character",
+    "java.lang.String",
+    "java.util.Date"
+]
+
+# To start CellProfiler with the plugins directory:
+#   `pythonw -m cellprofiler --plugins-directory .`
+class RunImageJ(cellprofiler.module.Module):
+    category = "Advanced"
+    module_name = "RunImageJ"
+    variable_revision_number = 1
+
+    def create_settings(self):
+        self._ij = None
+
+        self.host = cellprofiler.setting.Text(
+            "ImageJ server",
+            imagej.HOST
+        )
+
+        self.connect = cellprofiler.setting.DoSomething(
+            "",
+            "Connect",
+            self._connect
+        )
+
+        self.divider = cellprofiler.setting.Divider(u"———OUTPUTS———")
+
+        # These will get redefined after the module connects to the server.
+        self.ij_module = cellprofiler.setting.Choice(
+            "ImageJ module",
+            choices=["-- NONE --"]
+        )
+
+        self.input_details = []
+        self.input_settings = cellprofiler.setting.SettingsGroup()
+
+        self.output_details = []
+        self.output_settings = cellprofiler.setting.SettingsGroup()
+
+        self.input_count = cellprofiler.setting.HiddenCount([], "")
+
+    def display(self, workspace, figure):
+        image_count = len(workspace.display_data.images)
+        figure.set_subplots((image_count, 1))
+        for idx, image in enumerate(workspace.display_data.images):
+            figure.subplot_imshow(idx, 0, image)
+
+    def on_setting_changed(self, setting, pipeline):
+        if setting == self.ij_module:
+            self._create_ij_settings(setting.value)
+
+    def prepare_settings(self, setting_values):
+        self.host.value = setting_values[0]
+
+        # Create the settings needed, based on the module details.
+        self._connect(setting_values[3])
+
+        # Populate the values of the input settings and the output settings.
+        offset = 5
+        n_input_settings = int(setting_values[2])
+
+        for setting, value in zip(self.input_settings.settings, setting_values[offset:offset + n_input_settings]):
+            setting.value = value
+
+        for setting, value in zip(self.output_settings.settings, setting_values[offset + n_input_settings:]):
+            setting.value = value
+
+    def run(self, workspace):
+        # FIXME: Keep the original IDs in some data structure,
+        # so that we guarantee we run the correct module here.
+        id = self._ij.find(self.ij_module.value)[0]
```

Comment:
This shadows the builtin python `id` function, it might be a good idea to rename this.